internal/transform/format.go

```diff
@@ -0,0 +1,492 @@
+package transform
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"slices"
+
+	"github.com/pb33f/libopenapi"
+	v3 "github.com/pb33f/libopenapi/datamodel/high/v3"
+	"github.com/speakeasy-api/speakeasy-core/openapi"
+	"gopkg.in/yaml.v3"
+)
+
+// NodeType represents the type of YAML node structure we're dealing with
+type NodeType int
+
+const (
+	RootNode NodeType = iota
+	ComponentsNode
+	PathNode
+	InfoNode
+	OperationNode
+	SchemaNode
+	ResponseNode
+	SecurityNode
+	UrlNode
+	TagNode
+	UnknownNode
+)
+
+func (n NodeType) String() string {
```

Comment:
This looks brittle -- maybe better to just use a switch statement or a map

Reply:
Swapped to a switch